scripts/o3de/o3de/manifest.py

```diff
@@ -421,40 +431,48 @@ def filter_project_and_gem_templates_out(template_path,
 
     return list(filter(filter_project_and_gem_templates_out, get_all_templates()))
 
-
-def get_engine_json_data(engine_name: str = None,
-                         engine_path: str or pathlib.Path = None) -> dict or None:
-    if not engine_name and not engine_path:
-        logger.error('Must specify either a Engine name or Engine Path.')
+def get_json_data(object_typename: str = None,
+                  object_path: str or pathlib.Path = None,
+                  object_validator = callable) -> dict or None:
+    if not object_typename or not object_validator:
+        logger.error(f'Missing object info.')
+    
+    if not object_path:
+        logger.error(f'{object_typename} Path {object_path} has not been registered.')
         return None
 
-    if engine_name and not engine_path:
-        engine_path = get_registered(engine_name=engine_name)
-
-    if not engine_path:
-        logger.error(f'Engine Path {engine_path} has not been registered.')
+    object_path = pathlib.Path(object_path).resolve()
+    object_json = object_path / f'{object_typename}.json'
+    if not object_json.is_file():
+        logger.error(f'{object_typename} json {object_json} is not present.')
         return None
-
-    engine_path = pathlib.Path(engine_path).resolve()
-    engine_json = engine_path / 'engine.json'
-    if not engine_json.is_file():
-        logger.error(f'Engine json {engine_json} is not present.')
-        return None
-    if not validation.valid_o3de_engine_json(engine_json):
-        logger.error(f'Engine json {engine_json} is not valid.')
+    if not object_validator(object_json):
+        logger.error(f'{object_typename} json {object_json} is not valid.')
         return None
 
-    with engine_json.open('r') as f:
+    with object_json.open('r') as f:
         try:
-            engine_json_data = json.load(f)
+            object_json_data = json.load(f)
         except json.JSONDecodeError as e:
-            logger.warn(f'{engine_json} failed to load: {str(e)}')
+            logger.warn(f'{object_json} failed to load: {str(e)}')
```

Comment:
nit: you don't need a str(e) in an f-string. It implicitly uses an objects str() operator

Reply:
Thanks